server/src/main/java/io/druid/segment/loading/LocalDataSegmentPusher.java

```diff
@@ -78,7 +78,9 @@ public DataSegment push(File dataSegmentFile, DataSegment segment) throws IOExce
       );
     }
 
-    outDir.mkdirs();
+    if (!outDir.mkdirs() && !outDir.isDirectory()) {
```

Comment:
Like in the LocalDataSegmentPusher constructor?

I think it's good to avoid creating directories and stuff in constructors. We could put it in a `start` method but most of the other deep storages don't work that way, they usually defer creating anything until you call `push`.